### PR TITLE
Ex18 Miner Battle Royale

### PR DESCRIPTION
**Pull request by makcandrov**

### CODE REVIEW

File: src/Hack.sol [hunk 1]
```diff
@@ -843,3 +843,66 @@ contract Ticketing {
...
+/* Exercise 18 */
+
+/// @dev This contract represents a Battle Royale game where participants compete to become the "king" by achieving the lowest score.
+///      Participants can challenge the current king by deploying their own challenger contract.
+///      The challenger must return a non-empty response when called.
+///      If the challenger's address is lower than the current king's challenger address, they dethrone the previous king and become the new king.
+///      The rewards are distributed among the participants based on the time they held the king position.
+contract BattleRoyale {
+    uint256 public constant _duration = 1 weeks;
+    uint256 public constant _totalReward = 10 ether;
```
Comment (MerlinEgalite) [inline src/Hack.sol:856]: we put constant in capital letters though. 

And it's the internal, private state variables that we put the underscore sorry for not being clear enough 😆

File: src/Hack.sol [hunk 2]
```diff
@@ -843,3 +843,66 @@ contract Ticketing {
...
+
+        kingAddress = msg.sender;
+        kingChallenger = challenger;
+        dethronedTime = block.timestamp;
+
+        // If the user can't receive the reward, it will be burned.
+        previousKing.call{value: previousKingReward}("");
+    }
+
+    /// @dev Allows the current king to claim their reward at the end of the game.
```
Comment (MerlinEgalite) [inline src/Hack.sol:897]:
```suggestion
    /// @dev Allows the current king to claim its reward at the end of the game.
```

**Review by MerlinEgalite — CHANGES_REQUESTED** [review]
Not sure I get the issue please DM me haha

**Review by Tristan22400 — APPROVED** [review]

File: src/Hack.sol (hunk 2 above)
Comment (makcandrov) [reply src/Hack.sol:897]: You mean "his"? I think "their" is more inclusive here

**Review by MerlinEgalite — APPROVED** [review]

**Pull request merged**